### DISCUSSION TITLE
Can SLEIGH describe interpreted languages?

### DISCUSSION THREAD

**Discussion by HMaker** [NationalSecurityAgency/ghidra#7197] [Q&A] [answered]
Is there any example of SLEIGH for the Java Virtual Machine (Java), Common Language Runtime (.NET), cpython (Python) and many other interpreted languages?

**Comment by astrelsky** [comment, marked as answer]
https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Processors/JVM

**Comment by hippietrail** [comment]
I'm also, slowly, working on one for a 45-year-old Texas Instruments language called GPL that was used on the TI-99/4A home computer. It's not in [the repo](https://github.com/hippietrail/RetroGhidra) yet and I haven't started decompilation support.

There are projects out there for at least one of the [JavaScript engine's bytecode](https://github.com/PositiveTechnologies/ghidra_nodejs) and for [WebAssembly](https://github.com/garrettgu10/ghidra-wasm-plugin). The Ghidra folks have mentioned in the past that they have a CLR one that still hasn't been released yet.

**Reply by HMaker** [reply]
Thanks
